### DISCUSSION TITLE
Fields not considered to be used in Remix app

### DISCUSSION THREAD

**Discussion by oltodo** [0no-co/gql.tada#315] [Q&A] [answered]
Hi,
Sorry if the question already had been answered. 

I have a situation when I try to use gql.tada in a Remix application. In Remix's routes, the data are usually loaded into a [`loader` function](https://remix.run/docs/en/main/discussion/data-flow#route-loader) which is exported next to the component. So the GraphQL query is made outside the component, preventing gql.data from knowing whether the fields are being used.

![image](https://github.com/0no-co/gql.tada/assets/483842/d0470021-1645-4381-9152-c3a1a4fa3b84)

Do you have any ideas on how to get around the problem?

**Comment by kitten** [comment, marked as answer]
There's inherently no way to fix this, sorry! The easiest way of ensuring that this works as intended is either ignoring the issue or using fragments locally per your component where you're consuming the data.

The problem is that once you serialize the data the type becomes detached from the original definition site. When `json` is called, the type is basically remapped which means that GraphQLSP can't track it.

This is specific to Remix because they've got a type enforcement on their `json` utility.

- This is the option to disable field tracking: https://gql-tada.0no.co/reference/config-format#trackfieldusage
- This contains more details on fragment collocation if you want to detach field selection from your query: https://gql-tada.0no.co/guides/fragment-colocation

**Reply by oltodo** [reply]
Hi @kitten 
Thanks for your answer and explanations.
I'm probably going to deactivate it first.
